Userland/Libraries/LibC/netinet/ip.h

```diff
@@ -6,4 +6,32 @@
 
 #pragma once
 
+#include <assert.h>
+#include <bits/stdint.h>
+#include <sys/cdefs.h>
+
 #include "in.h"
+
+__BEGIN_DECLS
+
+struct [[gnu::packed]] ip {
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    uint8_t ip_v : 4;
+    uint8_t ip_hl : 4;
+#elif __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    uint8_t ip_hl : 4;
+    uint8_t ip_v : 4;
+#endif
+    uint8_t ip_tos;
+    uint16_t ip_len;
+    uint16_t ip_id;
+    uint16_t ip_off;
+    uint8_t ip_ttl;
+    uint8_t ip_p;
+    uint16_t ip_sum;
+    struct in_addr ip_src;
+    struct in_addr ip_dst;
+};
+static_assert(sizeof(struct ip) == 20, "struct ip: invalid length");
```

Comment:
Same C header comment, you could try `_Static_Assert()` or whatever that was, or just hide it in an ifdef

Reply:
There is a C definition for static_assert here, which I used:

https://github.com/SerenityOS/serenity/blob/dc9f516339be8850bbc0b4a1cc2051a72bf9b07a/Userland/Libraries/LibC/assert.h#L30

Happy to use _Static_assert if you want still, or is this ok?

Reply:
Ah, no worries then.
`static_assert` looks better, and works with C++ too~